WordPress/Classes/ViewRelated/Me/GravatarUploader.swift

```diff
@@ -0,0 +1,46 @@
+enum GravatarUploaderStatus {
+    case idle
+    case uploading(image: UIImage)
+    case finished
+}
+
+protocol GravatarUploader: AnyObject {
+    func updateGravatarStatus(_ status: GravatarUploaderStatus)
+}
+
+extension GravatarUploader {
+    func presentGravatarPicker(from sourceVC: UIViewController) {
+        WPAppAnalytics.track(.gravatarTapped)
+
+        let pickerViewController = GravatarPickerViewController()
+        pickerViewController.onCompletion = { [weak self] image in
+            if let updatedGravatarImage = image {
+                self?.uploadGravatarImage(updatedGravatarImage)
+            }
+
+            sourceVC.dismiss(animated: true, completion: nil)
+        }
+        pickerViewController.modalPresentationStyle = .formSheet
+        sourceVC.present(pickerViewController, animated: true, completion: nil)
+    }
+
+    func uploadGravatarImage(_ newGravatar: UIImage) {
+
+        let context = ContextManager.sharedInstance().mainContext
+        let accountService = AccountService(managedObjectContext: context)
+        guard let account = accountService.defaultWordPressComAccount() else {
+            return
+        }
+
+        WPAppAnalytics.track(.gravatarUploaded)
```

Comment:
Is the gravatar actually uploaded at this point? Would it make sense to do this later with `self?.updateGravatarStatus(.finished)`?

Reply:
It's not actually updated, but I thought I should leave it that way since that's how it was before. Thoughts?

Reply:
> I thought I should leave it that way since that's how it was before

Ya, I kinda figured as much. :) It's probably nitpicky. I could go either way.